File: spec/java/src/io/kaitai/struct/spec/TestDebugArrayUser.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.DebugArrayUser;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestDebugArrayUser extends CommonSpec {
    @Test
    public void testDebugArrayUser() throws Exception {
        DebugArrayUser r = DebugArrayUser.fromFile(SRC_DIR + "fixed_struct.bin");

        // --debug implies --no-auto-read
        r._read();

        assertIntEquals(r.oneCat().meow(), 80);
        assertIntEquals(r.arrayOfCats().size(), 3);
        assertIntEquals(r.arrayOfCats().get((int) 0).meow(), 65);
        assertIntEquals(r.arrayOfCats().get((int) 1).meow(), 67);
        assertIntEquals(r.arrayOfCats().get((int) 2).meow(), 75);
    }
}

File: spec/python/spec/test_debug_array_user.py
import unittest
from testformats.debug_array_user import DebugArrayUser

class TestDebugArrayUser(unittest.TestCase):
    def test_debug_array_user(self):
        with DebugArrayUser.from_file('src/fixed_struct.bin') as r:
            # --debug implies --no-auto-read
            r._read()

            self.assertEqual(r.one_cat.meow, 80)
            self.assertEqual(len(r.array_of_cats), 3)
            self.assertEqual(r.array_of_cats[0].meow, 65)
            self.assertEqual(r.array_of_cats[1].meow, 67)
            self.assertEqual(r.array_of_cats[2].meow, 75)
